pkg/vmprovider/providers/vsphere/vmprovider_vm_test.go

```diff
@@ -1494,6 +1494,98 @@ func vmTests() {
 						})
 					})
 				})
+
+				Context("Cloudinint Transport without userdata", func() {
+					var ec map[string]interface{}
+
+					JustBeforeEach(func() {
+						vm.Spec.VmMetadata = &vmopv1.VirtualMachineMetadata{
+							Transport: vmopv1.VirtualMachineMetadataCloudInitTransport,
+						}
+						vcVM, err := createOrUpdateAndGetVcVM(ctx, vm)
+						Expect(err).ToNot(HaveOccurred())
+
+						var o mo.VirtualMachine
+						Expect(vcVM.Properties(ctx, vcVM.Reference(), nil, &o)).To(Succeed())
+
+						ec = map[string]interface{}{}
+						for _, option := range o.Config.ExtraConfig {
+							if val := option.GetOptionValue(); val != nil {
+								ec[val.Key] = val.Value.(string)
+							}
+						}
```

Comment:
I feel like these should be executed inside the "It" block rather than "JustBeforeEach" here, so that you don't need to have the "AfterEach" block at all. Additionally, consider having a function to collect the `ec` (we might already have one defined as a  utility function) to avoid duplicate code.

Reply:
Moved inside "It" block. Thanks for the suggestion.